### PR TITLE
[Related #33] create button with input and its stories

### PR DESCRIPTION
**Pull request by supunTE**
Related #33 

- Create the button-with-input component
- Create stories for the component
- Update button and input components to catch events

### CODE REVIEW

File: apps/client/src/components/molecules/button/Button.component.tsx [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 
 export interface ButtonProps {
-  onClick: () => void;
+  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
   isCarVisible?: boolean;
   isDisable?: boolean;
   isButtonDanger?: boolean;
   children: string;
+  input?: ReactElement;
+  isFillWidth?: boolean;
```
Comment (dalanad) [inline apps/client/src/components/molecules/button/Button.component.tsx:13]: `isFullWidth`

File: apps/client/src/components/molecules/button/Button.component.tsx [hunk 2]
```diff
@@ -64,9 +71,12 @@ export function Button({
           }
         />
       ) : null}
-      <Text type='Label' size='Medium' className='truncate'>
-        {children}
-      </Text>
+      <div className='flex items-center flex-grow justify-center gap-10'>
+        {input ? input : null}
```
Comment (dalanad) [inline apps/client/src/components/molecules/button/Button.component.tsx:75]: Do we need the ternary operator here ?

**Review by dalanad — CHANGES_REQUESTED** [review]

File: apps/client/src/components/molecules/button/Button.component.tsx (hunk 2 above)
Comment (supunTE) [reply apps/client/src/components/molecules/button/Button.component.tsx:75]: Fixed in 689a160452b9e66a9cf28039c102bcb3248d76bb (PR #52)

File: apps/client/src/components/molecules/button/Button.component.tsx (hunk 1 above)
Comment (supunTE) [reply apps/client/src/components/molecules/button/Button.component.tsx:13]: Fixed in 9285214b129594bafcc9e8e797229d82d65898c7 (PR #52)

**Review by dalanad — APPROVED** [review]

**Review by pasinducw — APPROVED** [review]

**Pull request merged**